assets/src/block-validation/sidebar.js

```diff
@@ -12,22 +18,24 @@ import { useEffect } from '@wordpress/element';
 import './style.css';
 import AMPValidationErrorsIcon from '../../images/amp-validation-errors.svg';
 import AMPValidationErrorsKeptIcon from '../../images/amp-validation-errors-kept.svg';
+import { Loading } from '../components/loading';
 import { Error } from './error';
 import { BLOCK_VALIDATION_STORE_KEY } from './store';
-import { AMP_VALIDITY_REST_FIELD_NAME } from './constants';
 
 /**
  * Editor sidebar.
  */
 export function Sidebar() {
 	const { setIsShowingReviewed } = useDispatch( BLOCK_VALIDATION_STORE_KEY );
+	const { savePost } = useDispatch( 'core/editor' );
 
-	const { ampCompatibilityBroken, isShowingReviewed, status, reviewLink } = useSelect( ( select ) => ( {
+	const { ampCompatibilityBroken, isDirty, isFetchingErrors, isShowingReviewed, status, reviewLink } = useSelect( ( select ) => ( {
 		ampCompatibilityBroken: select( BLOCK_VALIDATION_STORE_KEY ).getAMPCompatibilityBroken(),
```

Comment:
Curious, for `getAMPCompatibilityBroken`, `getIsFetchingErrors`, `getIsShowingReviewed`, and `getReviewLink` I'm getting an IDE warning about a missing argument:

![image](https://user-images.githubusercontent.com/134745/105285636-a271e900-5b69-11eb-8638-4b2a32184471.png)

I guess that's erroneous warning since the store automatically provides it to the selectors?

Reply:
> I guess that's erroneous warning since the store automatically provides it to the selectors?

You're correct. The store always passes the state as the first argument to the selector. Any additional arguments are passed after that (e.g. [`getEditedPostAttribute`](https://github.com/WordPress/gutenberg/blob/0f09f1a38d82709c99292997d1b9decc4c9a9744/packages/editor/src/store/selectors.js#L373) function you can find below gets the `status` argument right after the `state`).

Btw. my IDE doesn't highlight this piece but it may be a configuration thing:

<img width="723" alt="Screenshot 2021-01-26 at 16 50 21" src="https://user-images.githubusercontent.com/478735/105869165-3c68e400-5ff7-11eb-897e-f95f9ea82210.png">